### PR TITLE
Removed ObserverSet dependency and switched to NotificationCenter

### PR DESCRIPTION
**Pull request by stephanheilner**
Xcode keeps showing that observer set isn't releasing object from memory. This it basically moving away from ObserverSet completely and using NotificationCenter instead.

### CODE REVIEW

**Comment by stephanheilner** [comment]
Xcode keeps showing that observer set isn't releasing object from memory. This it basically moving away from ObserverSet completely and using NotificationCenter instead.

**Comment by stephanheilner** [comment]
I also have an alternative solution that switches from `ObserverSet` to `Signals`

File: Theme/ThemeController.swift [hunk 1]
```diff
@@ -32,7 +31,7 @@ public class ThemeController: NSObject {
         didSet {
             theme?.setTheme(themeName: themeName)
 
-            themeObservers.notify(())
+            NotificationCenter.default.post(name: Notification.Name.ThemeDidChange, object: nil)
```
Comment (hiltonc) [inline Theme/ThemeController.swift]: I think you can just do `.ThemeDidChange` here and below.

Comment (hiltonc) [reply Theme/ThemeController.swift]: Also, pass `self` for `object` as that is the object sending the notification.

File: Theme/ThemeObserver.swift [hunk 2]
```diff
@@ -22,16 +22,19 @@
...
 
-extension ThemeObserver where Self: UIViewController {
-    public func observeTheme(_ f: @escaping (Self) -> () -> Void) {
-        ThemeController.shared.observeTheme(self, f)
+extension ThemeObserver {
+    public func observeTheme() {
+        NotificationCenter.default.addObserver(self, selector: #selector(themeDidChange), name: Notification.Name.ThemeDidChange, object: nil)
+    }
+    public func removeThemeObserver() {
+        NotificationCenter.default.removeObserver(self, name: Notification.Name.ThemeDidChange, object: nil)
```
Comment (hiltonc) [inline Theme/ThemeObserver.swift]: `.ThemeDidChange` here and above.

**Pull request closed without merging**